src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy

```diff
@@ -562,6 +562,24 @@ trait Address {
     }
   }
 
+  private Boolean isSubjectAddress(def entityAddressId, def entityTypeId = null, def entityId = null) {
+    if (notEmpty(entityTypeId) || notEmpty(entityId)) {
+      return isSubject(entityTypeId ?: entityId)
+    } else {
+      LinkedHashMap objAddress = getObjAddress(entityAddressId)
+      if (objAddress) {
+        return false
+      }
+
+      LinkedHashMap subjAddress = getSubjAddress(entityAddressId)
```

Comment:
okay, be more optimistic and check subj address at first like so:
```
LinkedHashMap subjAddress = getSubjAddress(entityAddressId)

return isPresent(subjAddress) || isEmpty(getObjAddress(entityAddressId)) || null
```

you don't need ```if```'s